lib/gruf/error.rb

```diff
@@ -130,16 +138,30 @@ def serialize
     end
 
     ##
-    # Append any appropriate errors to the gRPC call and properly update the output metadata
+    # Update the trailing metadata on the given gRPC call, including the error payload if configured
+    # to do so.
     #
     # @param [GRPC::ActiveCall] active_call The marshalled gRPC call
-    # @return [Error] Return the error itself with the GRPC::ActiveCall attached and error metadata appended
+    # @return [Error] Return the error itself after updating metadata on the given gRPC call.
+    #                 In the case of a metadata overflow error, we replace the current error with
+    #                 a new one that won't cause a low-level http2 error.
     #
     def attach_to_call(active_call)
       metadata[Gruf.error_metadata_key.to_sym] = serialize if Gruf.append_server_errors_to_trailing_metadata
-      if !metadata.empty? && active_call && active_call.respond_to?(:output_metadata)
-        active_call.output_metadata.update(metadata)
+      return self unless !metadata.empty? && active_call && active_call.respond_to?(:output_metadata)
```

Comment:
This is confusing to read. Prefer:

```ruby
return self if metadata.empty? || !active_call || !active_call.respond_to?(:output_metadata)
```

Reply:
Funny, I was going back and forth on this one. :)

Reply:
Yeah, I waffle on these cases generally too, but have settled into what seems to be industry best practice, which is in a conditional, if there are multiple conditions, prefer `if` over `unless` as that way you're not dealing with double negatives.